ble/services/DFUService.h

```diff
@@ -17,6 +17,8 @@
 #ifndef __BLE_DFU_SERVICE_H__
 #define __BLE_DFU_SERVICE_H__
 
+#ifdef TARGET_NRF51822 /* DFU only supported on nrf51 platforms */
```

Comment:
@rgrover @LiyouZhou: I am not sure what the convention is, but perhaps is better if we have something like `#if !defined(__BLE_DFU_SERVICE_H__) && defined(TARGET_NRF51822)` rather than:
```
#ifndef __BLE_DFU_SERVICE_H__
#define __BLE_DFU_SERVICE_H__

#ifdef TARGET_NRF51822
```